lib/User_Proxy.php

```diff
@@ -236,19 +236,19 @@ public function getDisplayName($uid) {
 	 *
 	 * @param string $uid the ownCloud user name
 	 * @return bool either the user can or cannot
-	 * @throws \OutOfBoundsException
-	 * @throws \InvalidArgumentException
-	 * @throws \BadMethodCallException
-	 * @throws ServerNotAvailableException
-	 * @throws DoesNotExistOnLDAPException
 	 */
 	public function canChangeAvatar($uid) {
-		foreach ($this->backends as $backend) {
-			if ($backend->userExists($uid)) {
-				return $backend->canChangeAvatar($uid);
+		try {
+			foreach ($this->backends as $backend) {
+				if ($backend->userExists($uid)) {
+					return $backend->canChangeAvatar($uid);
+				}
 			}
+		} catch (\Exception $e) {
```

Comment:
Is it worthy to log something, maybe for tracing purposes? Catching 4-5 exceptions (according to what has been deleted) seem to be a bunch of things that could go wrong, so we might want to know what happened. Maybe a debug message with the exception message is enough, we don't need to log the exception if we don't want to make too much noise (we're returning false already).